crates/red_knot_python_semantic/mdtest.py

```diff
@@ -0,0 +1,209 @@
+"""A runner for Markdown-based tests for Red Knot"""
+# /// script
+# requires-python = ">=3.11"
+# dependencies = [
+#     "rich",
+#     "watchfiles",
+# ]
+# ///
+
+from __future__ import annotations
+
+import json
+import os
+import subprocess
+from pathlib import Path
+from typing import Final, Literal, Never, assert_never
+
+from rich.console import Console
+from watchfiles import Change, watch
+
+CRATE_NAME: Final = "red_knot_python_semantic"
+CRATE_ROOT: Final = Path(__file__).resolve().parent
+MDTEST_DIR: Final = CRATE_ROOT / "resources" / "mdtest"
+
+
+class MDTestRunner:
+    mdtest_executable: Path | None
+    console: Console
+
+    def __init__(self) -> None:
+        self.mdtest_executable = None
+        self.console = Console()
+
+    def _run_cargo_test(self, message_format: Literal["human", "json"]) -> str:
```

Comment:
this parameter seems to always be passed using keyword arguments, which I think is more readable; you could possibly enforce it in the signature
```suggestion
    def _run_cargo_test(self, *, message_format: Literal["human", "json"]) -> str:
```